example/cancel_request.dart

```diff
@@ -34,14 +34,14 @@ FutureOr<void> main(List<String> args) async {
     ignoreThisFuture.ignore();
 
     final resp = await helloRespFuture;
-    print('/hello response: ${resp?.payloadString}');
+    print('/hello response: ${resp.payloadString}');
 
     if (cancelThisReq.retransmits > 0) {
       print('Expected 0 retransmits!');
     }
-
-    client.close();
   } catch (e) {
     print('CoAP encountered an exception: $e');
   }
+
+  client.close();
```

Comment:
I moved the close method below the try-catch-blocks in the examples, as otherwise the client is not closed when an exception is thrown (and caught).